CONTRIBUTING.md

```diff
@@ -1,3 +1,112 @@
+## Commits
+
+Please commit your changes frequently in small logical chunks that are 
+consistent, work independently of any later commits, and pass the linter as well 
+as the test suite. Doing so eases rollback and rebase operations. 
+
+You should also follow our commit message formatting rules, as they provide a 
+framework to write explicit messages that are easy to comprehend when looking 
+through the project history and enable automatic change log generation.
+
+These Guidelines were written based on 
+[AngularJS Git Commit Message Conventions](https://goo.gl/27wkkO).
+
+### Commit-Message
+
+Each commit message should consist of a header (type, scope, subject), a body 
+and a footer separated by empty lines:
+
+```
+<type>(<scope>): <subject>
+    
+<message>
+
+<footer>
+``` 
+
+Any line of the commit message must not be longer than 100 characters to ensure 
+that the messages are easy to read.
+
+### Subject
+
+The subject contains a succinct description of the change. It should use the 
+imperative and present tense; “change” not “changed” nor “changes”. 
+Don't capitalize the first letter, and don't end it with a dot.
+
+### Type
+
+The following commit types are allowed:
+
+* **feat** - 
+  use this type for commits that introduce a new features or capabilities
+* **fix** - use this one for bug fixes
+* **perf** - use this type for performance improvements
+* **docs** - use this one to indicate documentation adjustments and improvements 
+* **chore** - use this type for _maintainance_ commits e.g. removing old files 
+* **style** - use this one for commits that fix formatting and linting errors
+* **refactor** - 
+  use this type for adjustments to improve maintainability or performance
+* **test** - use this one for commits that add new tests
+
+### Scope
+
+The scope can be anything specifying the place of the committed change. 
```

Comment:
We've discussed to use filename when it's a single file change and module/package/page name for features etc.
We could expand this a little bit based on this discussion